llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/llama_index/vector_stores/kdbai/base.py

```diff
@@ -0,0 +1,214 @@
+"""KDB.AI vector store index.
+
+An index that is built within KDB.AI.
+
+"""
+
+import logging
+from typing import Any, List, Callable, Optional
+
+import pandas as pd
+
+from llama_index.core.bridge.pydantic import PrivateAttr
+from llama_index.core.schema import BaseNode, TextNode
+from llama_index.core.vector_stores.types import (
+    BasePydanticVectorStore,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+from .utils import default_sparse_encoder
+
+DEFAULT_COLUMN_NAMES = ["document_id", "text", "embedding"]
+
+DEFAULT_BATCH_SIZE = 100
+
+
+# INITIALISE LOGGER AND SET FORMAT
+logger = logging.getLogger(__name__)
+
+
+# MATCH THE METADATA COLUMN DATA TYPE TO ITS PYTYPE
+def convert_metadata_col(column, value):
+    try:
+        if column["pytype"] == "str":
+            return str(value)
+        elif column["pytype"] == "bytes":
+            return value.encode("utf-8")
+        elif column["pytype"] == "datetime64[ns]":
+            return pd.to_datetime(value)
+        elif column["pytype"] == "timedelta64[ns]":
+            return pd.to_timedelta(value)
+        return value.astype(column["pytype"])
+    except Exception as e:
+        logger.error(
+            f"Failed to convert column {column['name']} to type {column['pytype']}: {e}"
+        )
+
+
+class KDBAIVectorStore(BasePydanticVectorStore):
+    """The KDBAI Vector Store.
+
+    In this vector store we store the text, its embedding and
+    its metadata in a KDBAI vector store table. This implementation
+    allows the use of an already existing table.
+
+    Args:
+        table kdbai.Table: The KDB.AI table to use as storage.
+        batch (int, optional): batch size to insert data.
+            Default is 100.
+
+    Returns:
+        KDBAIVectorStore: Vectorstore that supports add and query.
+    """
+
+    stores_text: bool = True
+    flat_metadata: bool = True
+
+    hybrid_search: bool = False
+    batch_size: int
+
+    _table: Any = PrivateAttr()
+    _sparse_encoder: Optional[Callable] = PrivateAttr()
+
+    def __init__(
+        self,
+        table: Any = None,
+        hybrid_search: bool = False,
+        sparse_encoder: Optional[Callable] = None,
+        batch_size: int = DEFAULT_BATCH_SIZE,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize params."""
+        try:
+            import kdbai_client as kdbai
```

Comment:
this should be added as a dependency in the pyproject.toml

Reply:
As far as i can tell this is already included.  Did you mean somewhere else?
* https://github.com/run-llama/llama_index/blob/6f04122bc89e8e96afe2da198238219a72496cc3/llama-index-integrations/vector_stores/llama-index-vector-stores-kdbai/pyproject.toml#L39

Reply:
Oh I must have missed that